### PR TITLE
[FLINK-5524] [table] Support early out for code generated AND/OR condition

### PR DESCRIPTION
**Pull request by KurtYoung**
For condition like a AND b, if the result of a is false, we can save b from execution.

For condition like a OR b, if the result of a is true, we can also save b from execution.

### CODE REVIEW

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/CalcITCase.scala [hunk 1]
```diff
@@ -217,6 +218,65 @@ class CalcITCase(
   }
 
   @Test
+  def testDisjunctivePredicateEarlyOut(): Unit = {
```
Comment (twalthr) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/CalcITCase.scala]: Can you convert this test to a unit test with `ExpressionTestBase`? You can add them to `ScalarOperatorsTest`.

Comment (KurtYoung) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/CalcITCase.scala]: Sure, thanks for the hint.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/CalcITCase.scala [hunk 2]
```diff
@@ -217,6 +217,21 @@ class CalcITCase(
   }
 
   @Test
+  def testConjunctivePredicate(): Unit = {
```
Comment (twalthr) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/CalcITCase.scala]: This is basically a pure code generation change, we really need no ITCase for that. Can you remove this one? We really should keep ITCases to a minimum. They add to much overhead.

Comment (KurtYoung) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/CalcITCase.scala]: ok, will do.

**Comment by KurtYoung** [comment]
hey @twalthr, any more comments or this can be merged

**Comment by twalthr** [comment]
Thanks for the update @KurtYoung. The code looks good. I will merge this.

**Pull request closed without merging**